presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -558,6 +566,51 @@ protected Node visitShowSession(ShowSession node, Void context)
                     identifier("include"));
         }
 
+        private void checkObjectAccessible(Node node, QualifiedObjectName objectName)
+        {
+            if (!isCatalogAccessible(objectName.getCatalogName())) {
+                throw new SemanticException(MISSING_SCHEMA, node, "Catalog '%s' does not exist", objectName.getCatalogName());
+            }
+
+            if (!isSchemaAccessible(objectName.asCatalogSchemaName())) {
+                throw new SemanticException(MISSING_SCHEMA, node, "Schema '%s' does not exist", objectName.getSchemaName());
+            }
+
+            if (!isViewAccessible(objectName) && !isTableAccessible(objectName)) {
+                throw new SemanticException(MISSING_TABLE, node, "Table '%s' does not exist", objectName);
+            }
+        }
+
+        private boolean isCatalogAccessible(String catalogName)
+        {
+            return accessControl.filterCatalogs(session.getIdentity(), ImmutableSet.of(catalogName)).contains(catalogName)
+                    && metadata.catalogExists(session, catalogName);
```

Comment:
i would reverse -- check access control only for existing catalogs. I think an access control impl may have problems for non-existing catalogs.

(here & below)